src/main/java/uk/gov/hmcts/reform/ccd/documentam/configuration/SecurityConfiguration.java

```diff
@@ -1,90 +1,98 @@
 
-    private final AuthCheckerServiceAndUserFilter authCheckerFilter;
+    @Value("${spring.security.oauth2.client.provider.oidc.issuer-uri}")
+    private String issuerUri;
 
-    @Autowired
-    public SecurityConfiguration(final  RequestAuthorizer<User> userRequestAuthorizer,
-                                       final RequestAuthorizer<Service> serviceRequestAuthorizer,
-                                       final AuthenticationManager authenticationManager) {
-        super();
+    @Value("${oidc.issuer}")
+    private String issuerOverride;
+
+    private final ServiceAuthFilter serviceAuthFilter;
+    private final JwtAuthenticationConverter jwtAuthenticationConverter;
 
-        this.authCheckerFilter = new AuthCheckerServiceAndUserFilter(serviceRequestAuthorizer, userRequestAuthorizer);
+    private static final String[] AUTH_ALLOWED_LIST = {
+        "/swagger-resources/**",
+        "/swagger-ui/**",
+        "/webjars/**",
+        "/v2/api-docs",
+        "/health",
+        "/health/liveness",
+        "/health/readiness",
+        "/info",
+        "/favicon.ico",
+        "/"
+    };
 
-        this.authCheckerFilter.setAuthenticationManager(authenticationManager);
+    @Autowired
+    public SecurityConfiguration(ServiceAuthFilter serviceAuthFilter,
+                                 JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverter) {
+        super();
+        this.serviceAuthFilter = serviceAuthFilter;
+        jwtAuthenticationConverter = new JwtAuthenticationConverter();
+        jwtAuthenticationConverter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter);
     }
 
     @Override
     public void configure(WebSecurity web) {
-        web.ignoring().antMatchers("/swagger-ui.html",
-                                   "/webjars/springfox-swagger-ui/**",
-                                   "/swagger-resources/**",
-                                   "/v2/**",
-                                   "/health",
-                                   "/health/liveness",
-                                   "/status/health",
-                                   "/loggers/**",
-                                   "/");
+        web.ignoring().antMatchers(AUTH_ALLOWED_LIST);
     }
 
     @Override
-    protected void configure(final HttpSecurity http) throws Exception {
-        // Don't erase user credentials as this is needed for the user profile
-        final ProviderManager authenticationManager = (ProviderManager) authenticationManager();
-        authenticationManager.setEraseCredentialsAfterAuthentication(false);
-        authCheckerFilter.setAuthenticationManager(authenticationManager());
-
+    protected void configure(HttpSecurity http) throws Exception {
         http
-            .addFilter(authCheckerFilter)
+            .addFilterBefore(serviceAuthFilter, BearerTokenAuthenticationFilter.class)
             .sessionManagement().sessionCreationPolicy(STATELESS).and()
             .csrf().disable()
```

Comment:
This line is highlighted as a HIGH security risk by Sonar (https://sonarcloud.io/project/security_hotspots?id=ccd-case-document-am-api&hotspots=AXKlIBXrHtDcdm-qOd7q). Is there a simple solution to resolve this or should we park it for now?

Reply:
Need to look into that later> It's same security config we have in our other apps. Probably should address across ccd.